### DISCUSSION TITLE
How to connect llmware to azure open ai embedding

### DISCUSSION THREAD

**Discussion by abhijeets765** [llmware-ai/llmware#544] [Q&A]
Hello,

I am not getting any way to connect azure open ai embedding in llmware i can see option is there only for open ai api key and its embedding . I want to use azure open ai embedding and llm please let me know if anyone has idea on thiis

**Comment by 0xshubhs** [comment]
1. **Understand Azure OpenAI Embedding**: Learn how it works, what it gives you, and how to use it.

2. **Check LLMWare Compatibility**: Make sure LLMWare can handle custom embedding sources.

3. **Integrate Them**: If it's possible, connect Azure OpenAI Embedding with LLMWare.

4. **Secure Access**: Get the right permissions (like an API key) to connect securely to Azure.

5. **Decide on Data**: Figure out what text you want to send for embedding and how you'll use the results in LLMWare.

6. **Test and Fix**: Try it out, find any issues, and fix them. It's like checking a new gadget to make sure it works.

7. **Make it Efficient**: Ensure your setup can handle the workload well, especially if you're using it a lot.

That should cover the bases! If you need more help, just ask.
